src/functional/otf/binding/cpp_interface.py

```diff
@@ -79,13 +80,28 @@ def render_python_type(python_type: Type) -> str:
     return _TYPE_MAPPING[python_type]
 
 
-def _render_function_param(
-    param: interface.ScalarParameter | interface.BufferParameter, index: int
-) -> str:
-    if isinstance(param, interface.ScalarParameter):
-        return f"{render_python_type(param.scalar_type.type)} {param.name}"
+def render_scalar_type(scalar_type: ts.ScalarType) -> str:
+    if scalar_type.kind == ts.ScalarKind.BOOL:
+        return "bool"
+    elif scalar_type.kind == ts.ScalarKind.INT32:
+        return "int32_t"
+    elif scalar_type.kind == ts.ScalarKind.INT64:
+        return "int64_t"
+    elif scalar_type.kind == ts.ScalarKind.FLOAT32:
+        return "float"
+    elif scalar_type.kind == ts.ScalarKind.FLOAT64:
+        return "double"
     else:
+        raise ValueError("unsupported scalar type")
```

Comment:
I didn't invest too much into exceptions because I want to clean the whole thing up next cycle. Anyways, you have a point that even clean-up is easier if the messages are better to begin with, so I added some better messages.